pycaret/internal/pycaret_experiment/class_reg_experiment.py

```diff
@@ -429,52 +440,88 @@ def setup(
                         )
 
                     if isinstance(categorical_iterative_imputer, str):
-                        if categorical_iterative_imputer not in regressors:
+                        if categorical_iterative_imputer not in classifiers:
                             raise ValueError(
                                 "Invalid value for the categorical_iterative_imputer "
                                 "parameter, got {categorical_iterative_imputer}. "
-                                f"Allowed estimators are: {', '.join(regressors)}."
+                                f"Allowed estimators are: {', '.join(classifiers)}."
                             )
-                        categorical_iterative_imputer = regressors[
+                        categorical_iterative_imputer = classifiers[
                             categorical_iterative_imputer
-                        ].class_def()
+                        ].class_def(**classifiers[categorical_iterative_imputer].args)
                     elif not hasattr(categorical_iterative_imputer, "predict"):
                         raise ValueError(
                             "Invalid value for the categorical_iterative_imputer "
                             "parameter. The provided estimator does not adhere "
                             "to sklearn's API."
                         )
 
-                    num_estimator = TransfomerWrapper(
+                    categorical_indices = [
```

Comment:
Why do we get the column indices when we could just pass the names directly to ColumnTransformer? It feels to me like an extra, unnecessary, step.